src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -550,6 +542,28 @@ protected Model createFlattenedPom( File pomFile ) throws MojoExecutionException
         ModelBuildingRequest buildingRequest = createModelBuildingRequest( pomFile );
         Model effectivePom = createEffectivePom( buildingRequest, isEmbedBuildProfileDependencies(), this.flattenMode );
 
+        // parse dependency version in <profile>
+        Dependencies managedDependencies = new Dependencies();
+        if (effectivePom.getDependencyManagement() != null
```

Comment:
think about whether this logic should be applied to the input of the transformation taht's to come, or fixing the transformation code itself (e..g, createCleanPom, or at the end, after the copy).  in a way, this is changing the value to the source input of the transformation, as oppose to changing the transformation code itself and applying the fixes at appropriate places.

Reply:
update: have consolidated two blocks of code change into createCleanPom method